controllers/lmes/lmevaljob_controller.go

```diff
@@ -510,18 +518,20 @@ func (r *LMEvalJobReconciler) deleteJobPod(ctx context.Context, job *lmesv1alpha
 func (r *LMEvalJobReconciler) handleComplete(ctx context.Context, log logr.Logger, job *lmesv1alpha1.LMEvalJob) (ctrl.Result, error) {
 	if job.Status.CompleteTime == nil {
 		// make sure the pod is in the complete state. if not, run the shutdown command
-		pod, err := r.getPod(ctx, job)
-		if err == nil {
-			if getRunningContainerByName(&pod.Status, "main") != -1 {
-				// send shutdown command if the main container is running
-				if err := r.shutdownDriver(ctx, job); err != nil {
-					log.Error(err, "failed to shutdown the job pod. retry after the pulling interval")
-					return r.pullingJobs.addOrUpdate(string(job.GetUID()), Options.PodCheckingInterval), nil
+		if job.Status.PodName != "" {
```

Comment:
with this condition check, the handleComplete function can properly handle the case where the job pod hasn't been created yet. nice